FrameworkSystem/Client/MonitoringClient.py

```diff
@@ -224,64 +263,68 @@ def addMark( self, name, value = 1 ):
       # raise Exception( "Value's type %s is not valid" % value )
     self.activitiesLock.acquire()
     try:
-      self.logger.debug( "Adding mark to %s" % name )
-      markTime = self.__UTCStepTime( name )
-      if markTime in self.activitiesMarks[ name ]:
-        self.activitiesMarks[ name ][ markTime ].append( value )
+      self.logger.debug("Adding mark to %s" % name)
+      markTime = self.__UTCStepTime(name)
+      if markTime in self.activitiesMarks[name]:
+        self.activitiesMarks[name][markTime].append(value)
       else:
-        self.activitiesMarks[ name ][ markTime ] = [ value ]
+        self.activitiesMarks[name][markTime] = [value]
     finally:
       self.activitiesLock.release()
 
-  def __consolidateMarks( self, allData ):
+  def __consolidateMarks(self, allData):
     """
       Copies all marks except last step ones
-      and consolidates them
+      and consolidates them.
     """
     consolidatedMarks = {}
     remainderMarks = {}
     for key in self.activitiesMarks:
       if allData:
-        lastStepToSend = int( Time.toEpoch() )
+        lastStepToSend = int(Time.toEpoch())
       else:
-        lastStepToSend = self.__UTCStepTime( key )
-      consolidatedMarks[ key ] = {}
-      remainderMarks [ key ] = {}
-      for markTime in self.activitiesMarks[ key ]:
-        markValue = self.activitiesMarks[ key ][ markTime ]
+        lastStepToSend = self.__UTCStepTime(key)
+      consolidatedMarks[key] = {}
+      remainderMarks[key] = {}
+      for markTime in self.activitiesMarks[key]:
+        markValue = self.activitiesMarks[key][markTime]
         if markTime >= lastStepToSend:
-          remainderMarks[ key ][ markTime ] = markValue
+          remainderMarks[key][markTime] = markValue
         else:
-          consolidatedMarks[ key ][ markTime ] = markValue
+          consolidatedMarks[key][markTime] = markValue
           # Consolidate the copied ones
           totalValue = 0
-          for mark in consolidatedMarks[ key ][ markTime ]:
+          for mark in consolidatedMarks[key][markTime]:
             totalValue += mark
-          if self.activitiesDefinitions[ key ][ 'type' ] == self.OP_MEAN:
-            totalValue /= len( consolidatedMarks[ key ][ markTime ] )
-          consolidatedMarks[ key ][ markTime ] = totalValue
-      if len( consolidatedMarks[ key ] ) == 0:
-        del( consolidatedMarks[ key ] )
+          if self.activitiesDefinitions[key]['type'] == self.OP_MEAN:
+            totalValue /= len(consolidatedMarks[key][markTime])
+          consolidatedMarks[key][markTime] = totalValue
+      if len(consolidatedMarks[key]) == 0:
+        del(consolidatedMarks[key])
     self.activitiesMarks = remainderMarks
     return consolidatedMarks
 
-  def flush( self, allData = False ):
+  def flush(self, allData=False):
+    """ This is the method which is called by the gMonitoringFlusher object in order to flush all the data
```

Comment:
That's a bit too detailed. You do not need to explain the internals of each and every object that is used inside the function. 
You need to explain what it does, and what are the params. 

Inside the code itself, if needs to be, you can detail what is going on